pkg/cvo/internal/operatorstatus.go

```diff
@@ -71,35 +72,80 @@ func (b *clusterOperatorBuilder) Do() error {
 		b.modifier(os)
 	}
 
-	return waitForOperatorStatusToBeDone(b.client, os)
+	return waitForOperatorStatusToBeDone(osPollInternal, osPollTimeout, b.client, os)
 }
 
 const (
 	osPollInternal = 1 * time.Second
 	osPollTimeout  = 1 * time.Minute
 )
 
-func waitForOperatorStatusToBeDone(client configclientv1.ClusterOperatorsGetter, co *configv1.ClusterOperator) error {
-	var lastCO *configv1.ClusterOperator
-	err := wait.Poll(osPollInternal, osPollTimeout, func() (bool, error) {
-		eos, err := client.ClusterOperators().Get(co.Name, metav1.GetOptions{})
+func waitForOperatorStatusToBeDone(interval, timeout time.Duration, client configclientv1.ClusterOperatorsGetter, expected *configv1.ClusterOperator) error {
+	var lastErr error
+	err := wait.Poll(interval, timeout, func() (bool, error) {
+		actual, err := client.ClusterOperators().Get(expected.Name, metav1.GetOptions{})
 		if err != nil {
```

Comment:
Hrm.  I guess this makes sense.

However, this highlighted that we need to be passing Context into this, and we need to abort the poll loop if context is closed.

Also, we need to use `PollImmediate`.